### PR TITLE
Use standart Cordova CLI workflow when could not find specified target

### PR DESCRIPTION
**Pull request by JiglioNero**

### CODE REVIEW

File: src/debugger/cordovaDebugSession.ts [hunk 1]
```diff
@@ -1275,29 +1275,52 @@ To get the list of addresses run "ionic cordova run PLATFORM --livereload" (wher
...
+
+        const useDefaultCLI = async () => {
+            this.outputLogger("Continue using standard CLI workflow.");
+            targetArgs = ["--verbose"];
+            const debuggableDevices = await adbHelper.getOnlineDevices();
+            if (debuggableDevices.length) {
+                launchArgs.target = debuggableDevices[0].id;
+            } else {
+                launchArgs.target = TargetType.Emulator;
+            }
```
Comment (etatanova) [inline src/debugger/cordovaDebugSession.ts:1295]: May we use the ternary operator here?
```suggestion
            launchArgs.target = debuggableDevices.length ? debuggableDevices[0].id : TargetType.Emulator; 
```

File: src/debugger/cordovaDebugSession.ts [hunk 2]
```diff
@@ -1275,29 +1275,50 @@ To get the list of addresses run "ionic cordova run PLATFORM --livereload" (wher
...
+                    }
+                    launchArgs.target = targetDevice.id;
+                } else {
+                   this.outputLogger(`Could not find debugable target '${launchArgs.target}'.`, true);
+                   useDefaultCLI();
+                }
+            } else {
+                targetArgs.push("--device");
+                if (!isDevice) {
+                    if ((await adbHelper.getOnlineDevices()).find((device) => device.id === launchArgs.target)) {
```
Comment (RedMickey) [inline src/debugger/cordovaDebugSession.ts:1310]: I think it's better to create a special function for this check.

**Review by RedMickey — APPROVED** [review]

File: src/debugger/cordovaDebugSession.ts [hunk 3]
```diff
@@ -1275,29 +1275,52 @@ To get the list of addresses run "ionic cordova run PLATFORM --livereload" (wher
...
-                targetArgs.push("--emulator", `--target=${targetDevice.id}`);
-                if (isEmulator && targetDevice.name) {
-                    launchScenariousManager.updateLaunchScenario(launchArgs, {target: targetDevice.name});
+
+        const useDefaultCLI = async () => {
+            this.outputLogger("Continue using standard CLI workflow.");
+            targetArgs = ["--verbose"];
+            const debuggableDevices = await adbHelper.getOnlineDevices();
+            // By default, if a target is not specified Cordova CLI uses the first online target from ‘adb devices’ list (launched emulators are placed after devices).
+            // For more information see https://github.com/apache/cordova-android/blob/bb7d733cdefaa9ed36ec355a42f8224da610a26e/bin/templates/cordova/lib/run.js#L57-L68
```
Comment (etatanova) [inline src/debugger/cordovaDebugSession.ts:1292]:
```suggestion
            // By default, if the target is not specified, Cordova CLI uses the first online target from ‘adb devices’ list (launched emulators are placed after devices).
            // For more information, see https://github.com/apache/cordova-android/blob/bb7d733cdefaa9ed36ec355a42f8224da610a26e/bin/templates/cordova/lib/run.js#L57-L68
```

File: src/utils/android/androidEmulatorManager.ts [hunk 4]
```diff
@@ -154,4 +156,11 @@ export class AndroidEmulatorManager extends VirtualDeviceManager {
         }
         return emulatorsList;
     }
+
+    public async isEmulatorTarget(target: string): Promise<boolean> {
+        if (target.toLowerCase().includes(TargetType.Emulator) || (await this.getVirtualDevicesNamesList()).includes(target)) {
+            return true;
+        }
+        return false;
```
Comment (etatanova) [inline src/utils/android/androidEmulatorManager.ts:164]:
```suggestion
        return target.toLowerCase().includes(TargetType.Emulator) || (await this.getVirtualDevicesNamesList()).includes(target);
```

File: src/utils/android/androidEmulatorManager.ts [hunk 5]
```diff
@@ -43,11 +43,13 @@ export class AndroidEmulatorManager extends VirtualDeviceManager {
...
             const onlineDevices = await this.adbHelper.getOnlineDevices();
             for (let i = 0; i < onlineDevices.length; i++) {
-                if (onlineDevices[i].id === target) {
-                    return { id: onlineDevices[i].id };
-                }
-                if (await this.adbHelper.getAvdNameById(onlineDevices[i].id) === target) {
-                    return { name: target, id: onlineDevices[i].id };
+                if (onlineDevices[i].type === DeviceType.AndroidSdkEmulator) {
+                    if (onlineDevices[i].id === target) {
+                        return {name: await this.adbHelper.getAvdNameById(target), id: target };
```
Comment (etatanova) [inline src/utils/android/androidEmulatorManager.ts:48]:
```suggestion
                        return { name: await this.adbHelper.getAvdNameById(target), id: target };
```

File: src/debugger/cordovaDebugSession.ts [hunk 6]
```diff
@@ -1275,29 +1275,52 @@ To get the list of addresses run "ionic cordova run PLATFORM --livereload" (wher
...
+            launchArgs.target = debuggableDevices.length ? debuggableDevices[0].id : TargetType.Emulator;
+        };
+
+        try {
+            if ((await androidEmulatorManager.isEmulatorTarget(launchArgs.target))) {
```
Comment (etatanova) [inline src/debugger/cordovaDebugSession.ts:1297]:
```suggestion
            if (await androidEmulatorManager.isEmulatorTarget(launchArgs.target)) {
```

**Review by etatanova — APPROVED** [review]

**Pull request merged**